src/app/public/tickets/tickets.component.ts

```diff
@@ -169,7 +169,52 @@ export class TicketsComponent {
 
           this.ngOnInit();
         });
-      }
+    }
+  }
+
+  protected createNameChangeOrder(totalFee: number, ticketsWithFeeDue: any, token: any): void {
+    console.log((ticketsWithFeeDue));
+    console.log(typeof(ticketsWithFeeDue))
+    const order = new Order({
+      paymentMethod: 'stripe',
+      paymentFee: 0, // TODO ask if we need this fee here
+      total: totalFee,
+      paymentToken: token.id,
+    });
+    //check tickets match id of user
+    Observable.forkJoin(ticketsWithFeeDue.map(ticket => this.ticketApi.checkTicket(ticket)))
+      .subscribe(tickets => {
```

Comment:
You can use `.flatMap` to save you nesting calls to observables in a callback-style, and have them behave a little more like promises.

```js
return Observable.forkJoin(map tickets...)
  .flatMap(() => this.orders.processNameChangeFee(order, undefined))
  .flatMap(() => Observable.forkJoin(ticketsWithFeeDue.flatMap(...))
  .subscribe(tickets => {...}, error => {... error sweet alert ...});
```

Otherwise if you have an error at `processNameChangeFee` it won't show up in the sweet alert. I'll refactor the backend calls tonight and change this code, just adding this comment for more info on [Observables](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html) and [rxjs](https://github.com/ReactiveX/rxjs) best practice 👍